### PR TITLE
ref(proj-config): Always include transaction name config

### PR DESCRIPTION
**Pull request by iker-barriocanal**
The flag `organizations:transaction-name-normalize` has been enabled at 100% for a while. This PR removes all usages of the flag, so that we can remove it.

### CODE REVIEW

File: src/sentry/relay/config/__init__.py [hunk 1]
```diff
@@ -47,7 +47,6 @@
 
 #: These features will be listed in the project config
 EXPOSABLE_FEATURES = [
-    "organizations:transaction-name-normalize",
```
Comment (jjbayer) [inline src/sentry/relay/config/__init__.py:50]: I haven't fully thought this through, but we might want to keep this feature flag exposed for external Relays that still check the feature flag.

**Review by jjbayer — CHANGES_REQUESTED** [review]
Pressing "Request changes" because of open question.

**Comment by jjbayer** [comment]
Two more observations:
1. We should also remove the global clusterer switch (can be a separate PR though): https://github.com/getsentry/sentry/blob/aeb5d691aa53698dd44ce5b78b183cc80ef187b0/src/sentry/conf/server.py#L1730-L1731
2. Regarding the `EXPOSED_FEATURES` and https://github.com/getsentry/getsentry/pull/10168, we should probably hard-code the feature flag to `True` for a while, like here: https://github.com/getsentry/sentry/blob/aeb5d691aa53698dd44ce5b78b183cc80ef187b0/src/sentry/conf/server.py#L1080-L1081. That way, we can remove all feature handlers but still support external Relays.

File: src/sentry/relay/config/__init__.py (hunk 1 above)
Comment (iker-barriocanal) [reply src/sentry/relay/config/__init__.py:50]: I don't think this is a major problem. If an external relay no longer gets this feature flag, there will be no transaction name scrubbing in that relay. However, the following relay in chain (i.e. PoP relay) will do it, so there's no problem wrt that. At some point, we must remove feature flags to stop them from growing indefinitely, so I think it's fine to expect users to upgrade relay in such edge cases.

File: src/sentry/relay/config/__init__.py [hunk 2]
```diff
@@ -49,7 +49,6 @@
 EXPOSABLE_FEATURES = [
     "projects:span-metrics-extraction",
     "organizations:transaction-name-mark-scrubbed-as-sanitized",
-    "organizations:transaction-name-normalize",
```
Comment (jjbayer) [inline src/sentry/relay/config/__init__.py:52]: I would keep this feature flag exposed, for older relay versions.

**Review by jjbayer — CHANGES_REQUESTED** [review]

File: src/sentry/relay/config/__init__.py (hunk 2 above)
Comment (TBS1996) [reply src/sentry/relay/config/__init__.py:52]: @jjbayer  iker replied to you about this here: https://github.com/getsentry/sentry/pull/47080#discussion_r1165500908 whats your thoughts on his feedback ?

File: src/sentry/relay/config/__init__.py (hunk 1 above)
Comment (jjbayer) [reply src/sentry/relay/config/__init__.py]: @iker-barriocanal metrics are extracted in external relays though. So when the external relay does not scrub the transaction, the harm is already done.

File: src/sentry/relay/config/__init__.py (hunk 2 above)
Comment (jjbayer) [reply src/sentry/relay/config/__init__.py:52]: Replied in old thread as to why I think its worth keeping this feature flag.

File: src/sentry/relay/config/__init__.py (hunk 1 above)
Comment (iker-barriocanal) [reply src/sentry/relay/config/__init__.py]: I'm not strongly opinionated here, so leaving them around sounds good to me. It'd be good to know what relays and their versions are to help make these decisions too.

**Comment by iker-barriocanal** [comment]
Decision was to not proceed with this PR.

**Pull request closed without merging**